### ISSUE TITLE
Incorrect error message when using multiprocessing managers

### ISSUE THREAD

**Issue by tg137** [microsoft/pyright#605]
**Describe the bug**
When using the `register` method to register a simple object with a `multiprocessing.BaseManager`, Pyright complains when trying to access the registered object through the manager.

This is incorrect behaviour as registering the object with the manager allows the object to be accessed through the manager. See [here](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.managers.BaseManager.register) for the documentation of the register method and [here](https://stackoverflow.com/a/39742800/6565341) for a full example of it in use. 

**To Reproduce**
A simple code snippet to reproduce is as follows:

```python
from multiprocessing.pool import Pool
from multiprocessing.managers import BaseManager
import time

class Counter:
    def __init__(self, count_to: int):
        self.count_to = count_to

    def run(self):
        for x in range(1, self.count_to + 1):
            time.sleep(1)
            print(x)

def main():
    # Setup processing pool
    pool_size = 1
    pool = Pool(pool_size)

    # Register Counter class with the BaseManager
    BaseManager.register('Counter', Counter)
    manager = BaseManager()
    manager.start()

    # Create a new counter object that's shared between processes
    # This is the line that Pyright complains about
    counter = manager.Counter(10)
    pool.apply_async(counter.run, ())
    pool.close()

    # Wait for processing in pools to complete
    pool.join()

if __name__ == "__main__":
    main()
```

**Expected behavior**
It would be expected that Pyright would figure out that this is a valid call and not show an error message as a result.

**VS Code extension or command-line**
I am using Pyright version 1.1.30 for VS Code on Windows 10.

**Additional Context**
The actual error message is as follows:
```
Cannot access member "Counter" for type "BaseManager"
  Member "Counter" is unknown Pyright
```

**Comment by erictraut** [comment]
I've confirmed that Pyright is doing the right thing here. It is correctly honoring the typeshed stub for multiprocessing/managers.pyi. 

The problem is that "register" is a dynamic behavior that can't be represented in static type information. The expression "manager.Counter" is not something that can be statically defined in the type stub.

I can offer a couple of suggestions here:
1. You could file a bug in the [typeshed repo](https://github.com/python/typeshed) and suggest that the BaseManager class in managers.pyi should have a general `__getattr__` method added. It would need to return an `Any` type. This would eliminate the error reported for the expression "manager.Counter", but it would mean that the result of this expression would have no type information (including no support for type completions). If you can get the typeshed team to fix this, let me know. I occasionally incorporate the latest version of typeshed stubs into pyright, and I can update it to include this change.
2. In the meantime, you can use a "# type: ignore" comment on that line to silence the error.
3. Alternatively, you can specify that `manager` has an `Any` type like this: `manager: Any = BaseManager()`
4. Or use `cast` to cast both `manager` and the result of the expression `manager.Counter(10)` to the right types: `counter = cast(Counter, cast(Any, manager).Counter(10))`

**Comment by tg137** [comment]
Thanks a lot for your detailed reply here, it makes a lot of sense now.

I'll have a more detailed look at the typeshed repo tomorrow and add a bug for this. I'll make sure to come back here if it ends up getting fixed to let you know (although it sounds as though you'd pull it in pretty quickly anyway!)

I appreciate your suggestions as well, I think in the meantime I'll be going down the casting route  so that I've still got the advantage of type completions.
